Errors in azure_monitor_logs sink when running check components

@nazar554 Just spotted these. Could you please take a look?

Thanks!

```
+ cargo check --tests --no-default-features --features sinks-azure_monitor_logs
    Checking vector v0.11.0 (/home/james/src/vector)
error[E0599]: no method named `sign_oneshot_to_vec` found for struct `openssl::sign::Signer<'_>` in the current scope
   --> src/sinks/azure_monitor_logs.rs:272:32
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                                ^^^^^^^^^^^^^^^^^^^ method not found in `openssl::sign::Signer<'_>`

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:13
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |             ^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:25
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:78
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                                                                              ^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: required by `config::_::_serde::export::Err`

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `vector`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error[E0599]: no method named `sign_oneshot_to_vec` found for struct `openssl::sign::Signer<'_>` in the current scope
   --> src/sinks/azure_monitor_logs.rs:272:32
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                                ^^^^^^^^^^^^^^^^^^^ method not found in `openssl::sign::Signer<'_>`

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:13
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |             ^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:25
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> src/sinks/azure_monitor_logs.rs:272:78
    |
272 |         let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
    |                                                                              ^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: required by `config::_::_serde::export::Err`

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: build failed
+ exit 255
xargs: sh: exited with status 255; aborting
make: *** [check-component-features] Error 124
```

--- a/src/sinks/azure_monitor_logs.rs
+++ b/src/sinks/azure_monitor_logs.rs
@@ -263,8 +263,9 @@ impl AzureMonitorLogsSink {
             len, CONTENT_TYPE, X_MS_DATE, rfc1123date, RESOURCE
         );
         let mut signer = sign::Signer::new(hash::MessageDigest::sha256(), &self.shared_key)?;
+        signer.update(string_to_hash.as_bytes())?;
 
-        let signature = signer.sign_oneshot_to_vec(string_to_hash.as_bytes())?;
+        let signature = signer.sign_to_vec()?;
         let signature_base64 = base64::encode_block(&signature);
 
         Ok(format!(

